daemon/create_unix.go

```diff
@@ -117,8 +116,11 @@ func (daemon *Daemon) populateVolume(ctx context.Context, c *container.Container
 		log.G(ctx).WithError(err).Debugf("can't copy data from %s:%s, to %s", c.ID, mnt.Destination, volumePath)
 		return errors.Wrapf(err, "failed to populate volume")
 	}
-	defer mnt.Cleanup(compatcontext.WithoutCancel(ctx))
-	defer cleanup(compatcontext.WithoutCancel(ctx))
+	defer func() {
+		ctx := context.WithoutCancel(ctx)
+		cleanup(ctx)
+		mnt.Cleanup(ctx)
+	}
```

Comment:
Grouped these into a single defer and made it share a single context; note that the order of calls is reversed here, because `defer`'s are executed in reverse order of which they're added; double check if this order is correct (or if the original one had them reversed?)

Reply:
The order is correct, `cleanup` should be called before `mnt.Cleanup`.